packages/next/src/server/app-render/encryption.ts

```diff
@@ -72,12 +72,30 @@ async function encodeActionBoundArg(actionId: string, arg: string) {
 // Encrypts the action's bound args into a string.
 export async function encryptActionBoundArgs(actionId: string, args: any[]) {
   const { clientModules } = getClientReferenceManifestForRsc()
+  const stack = new Error().stack!.split('\n').slice(2).join('\n')
+  let error: Error | undefined
 
   // Using Flight to serialize the args into a string.
   const serialized = await streamToString(
-    renderToReadableStream(args, clientModules)
+    renderToReadableStream(args, clientModules, {
+      onError(err) {
+        error = err instanceof Error ? err : new Error(String(err))
+        error.stack = stack
+      },
```

Comment:
hmm is it possible for this to be called multiple times? or does it stop after the first error? (maybe if one of the bound args was JSX and threw while being rendered but that error just ended up serialized in the payload?)

Reply:
an aggregate would be nice but idk how well we can show that in the overlay. first one sounds sensible to me, all of them would need to be fixed anyway

we could throw the first one and console.error the ones after, wouldn't that also make them appear?